### PR TITLE
fix: defer ssl import and add curl fallback for AppImage compatibility

### PR DESCRIPTION
**Pull request by rowsail**
## Problem

The KiCad 10 AppImage bundles `libcrypto.so.3` which can be incompatible with
the host system's `libssl.so.3`. On systems where the host requires
`OPENSSL_3.3.0` (e.g. Ubuntu 24.04+), Python's `import ssl` fails with:

```
ImportError: .../libcrypto.so.3: version 'OPENSSL_3.3.0' not found
(required by /usr/lib/x86_64-linux-gnu/libssl.so.3)
```

This crashes the entire plugin import chain because `api.py` imports `ssl` at
module level. Since `__init__.py` catches `ImportError` silently, the plugin
simply never appears in KiCad — with no error message shown to the user.

## Root Cause

`api.py` has `import ssl` at the top level and calls `_make_ssl_context()` at
module load time (`_SSL_CTX = _make_ssl_context()`). The import chain
`plugin.py → dialog.py → api.py → import ssl` fails before the plugin can
register itself.

## Fix

1. **Defer `import ssl`** — moved from module level to inside `_make_ssl_context()`
   and `_urlopen()`, so plugin registration succeeds even when ssl is broken.

2. **Lazy-initialize `_SSL_CTX`** — created on first network request instead of
   at import time, guarded by `_check_ssl_available()`.

3. **Add curl subprocess fallback** — when `ssl` is entirely unavailable,
   `_urlopen()` routes HTTPS requests through the system `curl` binary. This
   keeps the plugin fully functional.

## Testing

Tested with KiCad 10.0.0 AppImage on Ubuntu where the OpenSSL mismatch occurs:

- Plugin registers and appears in KiCad toolbar ✓
- `_check_ssl_available()` correctly returns `False` ✓
- curl fallback activates automatically ✓
- API calls (search, fetch component, fetch UUIDs) all succeed via curl ✓
- No behavior change on systems where ssl works normally ✓

## Changes

- `src/kicad_jlcimport/easyeda/api.py`: 1 file, +131 −4 lines

### CODE REVIEW

File: src/kicad_jlcimport/easyeda/api.py [hunk 1]
```diff
@@ -215,19 +326,35 @@ def _urlopen(req, timeout=30):
...
     """
+    if not _check_ssl_available():
+        # ssl module broken — use curl fallback
+        url = req.full_url if hasattr(req, "full_url") else str(req)
+        headers = dict(getattr(req, "headers", {})) if hasattr(req, "headers") else {}
+        if hasattr(req, "unredirected_hdrs"):
+            headers.update(req.unredirected_hdrs)
+        data = getattr(req, "data", None)
+        return _curl_fetch(url, headers=headers, data=data,
+                           timeout=timeout, insecure=_allow_unverified)
```
Comment (jvanderberg) [inline src/kicad_jlcimport/easyeda/api.py]: **Curl fallback bypasses `urllib.error` exception handling in callers.**

When using the curl path, `_urlopen` returns a `_CurlResponse` — but callers like `_get_json`, `search_components`, and `fetch_product_image` catch `urllib.error.HTTPError` and `urllib.error.URLError`. Those exceptions will never be raised by the curl path since `_curl_fetch` raises `APIError` directly for HTTP errors.

This currently works because:
- `_curl_fetch` already converts HTTP errors to `APIError`
- Callers that catch `urllib.error.*` also end up raising `APIError`

However, `fetch_product_image` catches those exceptions and returns `None` (graceful degradation). With the curl path, an HTTP error from `_curl_fetch` will instead raise `APIError` and propagate up — changing the behavior from "no image" to a hard failure. Consider wrapping the curl call in `fetch_product_image`'s `_urlopen` usage, or having `_curl_fetch` not raise on non-critical HTTP errors for image fetches.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by rowsail** [comment]
**All requested changes have been addressed across two follow-up commits:**

**Must-fix items:**

1. **`-L` flag added** — curl now follows redirects (`ccc5270`). Status code parsing also moved to stderr via `%{stderr}%{http_code}` to avoid conflating body and status data.

2. **`fetch_product_image` error behavior fixed** — Both `_urlopen` calls in `fetch_product_image` (and `download_step`, `download_wrl_source`) already catch `APIError` for graceful degradation. Added the same `except APIError: raise` guard to `_get_json` for consistency with `search_components` (`19e19c5`).

3. **Full test coverage added** — 782 tests passing, 94.78% overall coverage, `api.py` at 100%. Tests cover curl fallback (mocked `subprocess.run`), SSL availability check, lazy initialization, status code parsing edge cases, DNS caching, and the APIError passthrough path.

**Nice-to-have items:**

4. **Lazy `_CURL_PATH` resolution** — `shutil.which("curl")` now deferred to first use via `_get_curl_path()` instead of running at import time.

5. **Status code parsing safety** — Replaced the `rfind("\n")` approach entirely. Status code is now written to stderr (`-w "%{stderr}%{http_code}"`) and parsed from `proc.stderr`, completely separating it from the response body on stdout.
